### PR TITLE
[Merged by Bors] - chore: Do not import algebra in `Data.Finset.Lattice`

### PR DESCRIPTION
**Pull request by YaelDillies**
Move the `OrderBot Nat` instance to a new file `Order.Nat` and the `LinearOrderedSemiring` lemmas from `Data.Finset.Lattice` to a new file `Algebra.Order.Ring.Finset`. I credit Eric for https://github.com/leanprover-community/mathlib/pull/12912.

---

Note that @Ruben-VandeVelde will move more instances to `Order.Nat` soon.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Multiset/Fintype.lean [hunk 1]
```diff
@@ -119,7 +117,7 @@ theorem Multiset.mem_toEnumFinset (m : Multiset α) (p : α × ℕ) :
 #align multiset.mem_to_enum_finset Multiset.mem_toEnumFinset
 
 theorem Multiset.mem_of_mem_toEnumFinset {p : α × ℕ} (h : p ∈ m.toEnumFinset) : p.1 ∈ m :=
-  Multiset.count_pos.mp <| pos_of_gt <| (m.mem_toEnumFinset p).mp h
+  have :=  (m.mem_toEnumFinset p).mp h; Multiset.count_pos.mp (by omega)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Multiset/Fintype.lean:120]:
```suggestion
  have := (m.mem_toEnumFinset p).mp h; Multiset.count_pos.mp (by omega)
```

File: Mathlib/Order/Nat.lean [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/-!
+# The natural numbers form a linear order
+
+This file contains the linear order instance on the natural numbers.
+
+See note [foundational algebra order theory].
+
+## TODO
+
+Move the `LinearOrder ℕ` instance here.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Nat.lean:17]:
```suggestion
Move the `LinearOrder ℕ` instance here (#13082).
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
(Not fully through yet)

File: Mathlib/Data/Finset/Lattice.lean [hunk 3]
```diff
@@ -2076,33 +2078,6 @@ namespace Finset
 
 /-! ### Interaction with ordered algebra structures -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Finset/Lattice.lean:2080]:
```suggestion
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks! Please put on the maintainer queue with this comment addressed

**Comment by YaelDillies** [comment]
maintainer merge

(on behalf of Ruben)

**Comment by RemyDegenne** [comment]
bors d+
(not r+ since CI is still running)

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**